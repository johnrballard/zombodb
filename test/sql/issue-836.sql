CREATE TABLE issue836 (
    id serial8 not null primary key,
    b bool
);
insert into issue836 (id, b) values (default, false);
insert into issue836 (id, b) values (default, true);
create index idxissue836 on issue836 using zombodb ((issue836.*));

select * from zdb.highlight_document('issue836', (select row_to_json(issue836) from issue836 where b = true), 'b:true or b:false or b:TRUE or b:FALSE or b:NOT_a_Boolean', false);
select * from zdb.highlight_document('issue836', (select row_to_json(issue836) from issue836 where b = false), 'b:true or b:false or b:TRUE or b:FALSE or b:NOT_a_Boolean', false);

drop table issue836 cascade;