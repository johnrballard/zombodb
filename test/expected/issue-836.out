CREATE TABLE issue836 (
    id serial8 not null primary key,
    b bool
);
insert into issue836 (id, b) values (default, false);
insert into issue836 (id, b) values (default, true);
create index idxissue836 on issue836 using zombodb ((issue836.*));
select * from zdb.highlight_document('issue836', (select row_to_json(issue836) from issue836 where b = true), 'b:true or b:false or b:TRUE or b:FALSE or b:NOT_a_Boolean', false);
 field_name | array_index | term |   type    | position | start_offset | end_offset | query_clause 
------------+-------------+------+-----------+----------+--------------+------------+--------------
 b          |           0 | true | <BOOLEAN> |        1 |            0 |          0 | b:"true"
 b          |           0 | true | <BOOLEAN> |        1 |            0 |          0 | b:"true"
(2 rows)

select * from zdb.highlight_document('issue836', (select row_to_json(issue836) from issue836 where b = false), 'b:true or b:false or b:TRUE or b:FALSE or b:NOT_a_Boolean', false);
 field_name | array_index | term  |   type    | position | start_offset | end_offset | query_clause 
------------+-------------+-------+-----------+----------+--------------+------------+--------------
 b          |           0 | false | <BOOLEAN> |        1 |            0 |          0 | b:"false"
 b          |           0 | false | <BOOLEAN> |        1 |            0 |          0 | b:"false"
(2 rows)

drop table issue836 cascade;
